lib/core/logger.cpp

```diff
@@ -0,0 +1,107 @@
+#include "logger.hpp"
+
+#include <chrono>
+
+namespace hyped::utils {
+void Logger::setLevel(const Level level)
+{
+  level_ = level;
+};
+
+void Logger::intToLevel(const int level)
+{
+  switch (level) {
+    case 0:
+      level_ = Level::kNone;
+      break;
+    case 1:
+      level_ = Level::kDebug;
+      break;
+    case 2:
+      level_ = Level::kInfo;
+      break;
+    case 3:
+      level_ = Level::kWarn;
+      break;
+    case 4:
+      level_ = Level::kFatal;
+      break;
+    default:
+      level_ = Level::kNone;
+      break;
+  }
+};
+
+void Logger::printHead(FILE *file, const char *title)
+{
+  const std::time_t result = std::time(nullptr);
+  const tm *current_time        = localtime(&result);
```

Comment:
This is not how we usually get timestamps, right? This should use a time source instead. This might make it deterministic in testing, too?

Reply:
Will switch over to the constructor accepting a ITimeSource and using now().